core/src/main/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingService.scala

```diff
@@ -0,0 +1,150 @@
+    dataSource.inTransaction { dataAccess =>
+        dataAccess.workspaceQuery.listWithBillingProject(billingProjectName)
+    }.map { workspaces =>
+      workspaces.collect {
+        case workspace if workspace.workspaceVersion == WorkspaceVersions.V2 => GoogleProject(workspace.googleProjectId.value)
+      }.toSet
+    }
+  }
+
+  private def validateReportParameters(startDate: DateTime, endDate: DateTime): Unit = {
+    if (startDate.isAfter(endDate)) throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, "start date must be before end date"))
+    else if (Days.daysBetween(startDate, endDate).getDays > 90) throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, "provided dates exceed maximum report date range"))
+  }
+
+  def getSpendForBillingProject(billingProjectName: RawlsBillingProjectName, startDate: DateTime, endDate: DateTime): Future[Option[SpendReportingResults]] = {
+    validateReportParameters(startDate, endDate)
+
+    requireAlphaUser() {
+      requireProjectAction(billingProjectName, SamBillingProjectActions.readSpendReport) {
+        for {
+          spendExportConf <- getSpendExportConfiguration(billingProjectName)
+          workspaceProjects <- getWorkspaceGoogleProjects(billingProjectName)
+
+          query =
+          s"""
+             | SELECT
+             |  SUM(cost) as cost,
+             |  SUM(IFNULL((SELECT SUM(c.amount) FROM UNNEST(credits) c), 0)) as credits,
+             |  currency,
+             |  DATE(_PARTITIONTIME) as date
+             | FROM `${spendExportConf.spendExportTable.getOrElse(defaultTableName)}`
+             | WHERE billing_account_id = @billingAccountId
+             | AND _PARTITIONTIME BETWEEN @startDate AND @endDate
+             | AND project.id in UNNEST(@projects)
+             | GROUP BY currency, date
+             |""".stripMargin
+
+          queryParams = List(
+            new QueryParameter().setParameterType(new QueryParameterType().setType("STRING")).setName("billingAccountId").setParameterValue(new QueryParameterValue().setValue(spendExportConf.billingAccountId.withoutPrefix())),
+            new QueryParameter().setParameterType(new QueryParameterType().setType("STRING")).setName("startDate").setParameterValue(new QueryParameterValue().setValue(dateTimeToISODateString(startDate))),
+            new QueryParameter().setParameterType(new QueryParameterType().setType("STRING")).setName("endDate").setParameterValue(new QueryParameterValue().setValue(dateTimeToISODateString(endDate))),
+            new QueryParameter().setParameterType(new QueryParameterType().setType("ARRAY").setArrayType(new QueryParameterType().setType("STRING"))).setName("projects").setParameterValue(new QueryParameterValue().setArrayValues(workspaceProjects.map(project => new QueryParameterValue().setValue(project.value)).toList.asJava))
+          )
```

Comment:
Formatting ⛏️: This line is extremely long, anything we can do to improve the readability?